flaml/autogen/oai/exemplar_selector.py

```diff
@@ -0,0 +1,55 @@
+from functools import partial
+import random
+import numpy as np
+from flaml.autogen.oai.selection_methods import RandomSelect
+
+class ExemplarSelector:
+    METHOD_MAPPING = {
+        "random": RandomSelect,
+        # You can add more methods here...
+    }
+
+    @classmethod
+    def get_few_shot_template(cls, train_data, method=None, few_shot_template=None, method_params=None, template_params=None):
```

Comment:
Instead of passing selection method into exemplar_selector, maybe it would be easier to implement exemplar_selector with different methods?

```python
abstract class ExemplarSelector:
   def construct_template(context) -> str;
      pass

# random_exemplar_selector
class RandomExemplarSelector(ExemplarSelector):
    def __init__(self, train_data, k):
        # initialize

    def construct_template(context):
        # select `k` examples from train_data and render template. 
```

Reply:
I add the abstract class to exemplar_selector.py